dali/pipeline/data/sample_view_test.cc

```diff
@@ -53,13 +52,45 @@ TEST(SampleView, Constructors) {
 
   SampleView<CPUBackend> from_void_ptr{reinterpret_cast<void *>(42), {1, 2, 3}, DALI_FLOAT};
   compare(from_void_ptr, reinterpret_cast<void *>(42), {1, 2, 3}, DALI_FLOAT);
+
+  const int32_t cdata{};
+  ConstSampleView<CPUBackend> const_from_ptr{&cdata, {1, 2, 3}};
+  compare(const_from_ptr, &cdata, {1, 2, 3}, DALI_INT32);
+
+  ConstSampleView<CPUBackend> const_from_void_ptr{
+      reinterpret_cast<const void *>(42), {1, 2, 3}, DALI_FLOAT};
+  compare(const_from_void_ptr, reinterpret_cast<void *>(42), {1, 2, 3}, DALI_FLOAT);
+
+  ConstSampleView<CPUBackend> const_from_nonconst{from_ptr};
+  compare(const_from_nonconst, &data, {1, 2, 3}, DALI_INT32);
```

Comment:
For completeness - plain copy (for both const and non-const) and perhaps move.

Reply:
Done